File: web_viewer/backend/schemas/patient.py
"""Pydantic schemas for Patient endpoints."""

from datetime import date, datetime

from pydantic import BaseModel


class PatientCreate(BaseModel):
    """Request to create a new patient."""

    patient_code: str
    first_name: str | None = None
    last_name: str | None = None
    date_of_birth: date | None = None
    sex: str | None = None
    email: str | None = None
    phone: str | None = None
    medical_record_number: str | None = None
    insurance_provider: str | None = None
    insurance_policy_number: str | None = None
    notes: str | None = None
    metadata_json: str | None = None


class PatientUpdate(BaseModel):
    """Request to update a patient."""

    patient_code: str | None = None
    first_name: str | None = None
    last_name: str | None = None
    date_of_birth: date | None = None
    sex: str | None = None
    email: str | None = None
    phone: str | None = None
    medical_record_number: str | None = None
    insurance_provider: str | None = None
    insurance_policy_number: str | None = None
    notes: str | None = None
    metadata_json: str | None = None


class PatientRead(BaseModel):
    """Response schema for patient."""

    id: str
    patient_code: str
    first_name: str | None = None
    last_name: str | None = None
    date_of_birth: date | None = None
    sex: str | None = None
    email: str | None = None
    phone: str | None = None
    medical_record_number: str | None = None
    insurance_provider: str | None = None
    insurance_policy_number: str | None = None
    notes: str | None = None
    metadata_json: str | None = None
    created_at: datetime
    updated_at: datetime
    created_by: str

    model_config = {"from_attributes": True}

File: web_viewer/frontend/src/types/patient.ts
/**
 * Types for patient records in the doctor portal.
 */

export interface Patient {
  id: string;
  patient_code: string;
  first_name: string | null;
  last_name: string | null;
  date_of_birth: string | null;
  sex: string | null;
  email: string | null;
  phone: string | null;
  medical_record_number: string | null;
  insurance_provider: string | null;
  insurance_policy_number: string | null;
  notes: string | null;
  metadata_json: string | null;
  created_at: string;
  updated_at: string;
  created_by: string;
}

export interface PatientCreate {
  patient_code: string;
  first_name?: string;
  last_name?: string;
  date_of_birth?: string;
  sex?: string;
  email?: string;
  phone?: string;
  medical_record_number?: string;
  insurance_provider?: string;
  insurance_policy_number?: string;
  notes?: string;
  metadata_json?: string;
}

export interface PatientUpdate {
  patient_code?: string;
  first_name?: string;
  last_name?: string;
  date_of_birth?: string;
  sex?: string;
  email?: string;
  phone?: string;
  medical_record_number?: string;
  insurance_provider?: string;
  insurance_policy_number?: string;
  notes?: string;
  metadata_json?: string;
}
